include/exec/system_scheduler.hpp

```diff
@@ -0,0 +1,663 @@
+/*
+ * Copyright (c) 2023 Lee Howes
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "stdexec/execution.hpp"
+// For the default implementation, test will override
+#include "exec/static_thread_pool.hpp"
+
+#include <iostream>
+
+struct __exec_system_scheduler_interface;
+struct __exec_system_sender_interface;
+struct __exec_system_scheduler_impl;
+struct __exec_system_sender_impl;
+
+
+// Virtual interfaces to underlying implementations for initial simplicit
+// TODO: Potentially move to custom vtable implementations
+struct __exec_system_context_interface {
+  virtual __exec_system_scheduler_interface* get_scheduler() noexcept = 0;
+};
+
+// bulk function for scheduler to transmit from, will wrap actual function stub stored in real type
+using __exec_system_bulk_shape = long;
+using __exec_system_bulk_fn = void(void*, __exec_system_bulk_shape);
+struct __exec_system_bulk_function_object {
+  void* fn_state = nullptr;
```

Comment:
we should decide if we're ugly-ifying identifiers in this header or not and stick to it. I would prefer all identifiers to be ugly-ified.